### PR TITLE
feature: S3C-2785 add object lock check

### PR DESCRIPTION
**Pull request by ilkescality**
Adds objectLockEnabled check to Bucket info model.

Related: https://github.com/scality/cloudserver/pull/2510

### CODE REVIEW

**Review by tmacro — APPROVED** [review]

**Review by tmacro — CHANGES_REQUESTED** [review]
I believe that the model version needs to be bumped here
https://github.com/scality/Arsenal/blob/development/8.1/lib/constants.js

**Review by nicolas2bert — APPROVED** [review]

File: tests/unit/models/BucketInfo.js [hunk 1]
```diff
@@ -188,15 +190,16 @@ Object.keys(acl).forEach(
...
-                                      'string');
-               });
+            it('this should have the right BucketInfo types', () => {
+                assert.strictEqual(typeof dummyBucket.getName(), 'string');
+                assert.strictEqual(typeof dummyBucket.getOwner(), 'string');
+                assert.strictEqual(typeof dummyBucket.getOwnerDisplayName(),
+                                    'string');
+                assert.strictEqual(typeof dummyBucket.getCreationDate(),
+                                    'string');
+                assert.strictEqual(typeof dummyBucket.isObjectLockEnabled(),
```
Comment (dora-korpar) [inline tests/unit/models/BucketInfo.js:203]: Just wanna make sure this is what you meant for the test -> I'm pretty sure `objectLockEnabled` is defaulting to false, which will still allow the test to pass, but if you want to have it set to true, you have to set in in the instantiation for BucketInfo on line 149 :)

**Comment by dora-korpar** [comment]
@tmacro I'm bumping the model version in my arsenal PR that follows this one (https://github.com/scality/Arsenal/pull/995), and will update the readme in cloudserver as well

**Review by dora-korpar — APPROVED** [review]

**Review by tmacro — APPROVED** [review]

**Review by rahulreddy — CHANGES_REQUESTED** [review]
@ilkescality Can you please add another test that shows that a bucket created has object lock disabled as default?

**Comment by ilkescality** [comment]
@bert-e force-reset

**Comment by ilkescality** [comment]
@bert-e reset

**Comment by ilkescality** [comment]
@bert-e reset

**Review by rahulreddy — APPROVED** [review]

**Comment by rahulreddy** [comment]
@ilkescality  On 8.1 branch looks like you resolved the conflict in a way that `bucketPolicy` appears twice. Your forward port will pass if you fix that.

**Comment by ilkescality** [comment]
> @ilkescality On 8.1 branch looks like you resolved the conflict in a way that `bucketPolicy` appears twice. Your forward port will pass if you fix that.

@rahulreddy Thanks for bringing this to my attention; I didn't get bert-e email for failed CI due to `Argument name clash` and didn't realize it until seeing this message and checking `eve` manually, it's fixed & merged now.

**Pull request merged**